src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -362,6 +362,13 @@ ErrorIfUnsupportedCitusLocalTableKind(Oid relationId)
 							   "tables and foreign tables can be added to citus metadata ",
 							   relationName)));
 	}
+
+	if (get_rel_persistence(relationId) == RELPERSISTENCE_TEMP)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot add temporary table \"%s\" to citus metadata",
```

Comment:
Would this make sense in the context of adding a foreign key or a CREATE TABLE?  (as in, would the user know what the problem is and how to fix it?)

Reply:
>would the user know what the problem is

yeah, we catch the error messages in utility hook and give below detail in addition to the error message that we give here
https://github.com/citusdata/citus/blob/cacb76d2c6a70383ad5d7c597daa2298e446ac2b/src/backend/distributed/commands/table.c#L870..L874

>and how to fix it?

Maybe I should give a hint for that:
`HINT: Consider using persistent or unlogged tables`

Here I wish to tell `or drop the foreign key` to provide another alternative but since this is `CreateCitusLocalTable` function, it doesn't seem much reasonable/possible.

What do do you think @marcocitus ?

Reply:
actually, you cannot create foreign keys from temporary table to regular tables at all, so we should just let it fall through / give the same error as postgres.
```sql
postgres=# create table ref (a int primary key);
CREATE TABLE
postgres=# create temp table tmp (a int references ref (a));
ERROR:  constraints on temporary tables may reference only temporary tables
```

Reply:
Makes sense, then I guess I should give this error around `ConvertPostgresLocalTablesToCitusLocalTables` (not in `CreateCitusLocalTable`) since `CreateCitusLocalTable` might not be the right place to mention "constraints on temporary tables" ?

Reply:
(or for simplicity, let's give the same error in `CreateCitusLocalTable`)

Reply:
cc38e48